src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```diff
@@ -78,9 +84,18 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (config.tryGetProject() == null || config.autoServiceAccountUnavailable()) {
       return;
     }
-    // validate schema with underlying table
-    Schema schema = configuredSchema == null ? inputSchema : configuredSchema;
-    validateConfiguredSchema(schema, collector);
+
+    // Try validations which may require connectivity
+    try {
+      // Validate partition properties
+      config.validatePartitionProperties(inputSchema, configuredSchema, collector);
+      // validate schema with underlying table
+      Schema schema = configuredSchema == null ? inputSchema : configuredSchema;
+      validateConfiguredSchema(schema, collector);
+    } catch (InvalidStageException e) {
```

Comment:
this confuses me, did something get changed to throw this exception?

Reply:
So originally, there were two overloaded versions of the getBigQueryTable function in BigQueryUtils.java which are called in various places. One version takes a FailureCollector and one does not. Since calling FailureCollector causes an immediate failure, we needed to remove the overloaded method which calls FailureCollector and instead catch the error somewhere above the callstack. Thus, the FailureCollector overload was removed in favor of the other method, which instead throws InvalidConfigPropertyException and InvalidStageException.